internal/aez/interpolate.go

```diff
@@ -78,12 +51,16 @@ func Interpolate(
 	x float64, xSlice []float64, ySlice []float64,
 ) (float64, error) {
 	if len(xSlice) != len(ySlice) {
-		return -9999, ErrXYMismatch{
+		log.Fatalf(
```

Comment:
we don't want to halt execution on this I think, but produce an error. Fatal goes into panic if I understand things correctly and then you need to unpanic and produce an error instead.